addons/test_mail/tests/test_mail_activity_plan.py

```diff
@@ -86,6 +86,31 @@ def setUpClass(cls):
             } for idx in range(5)
         ])
 
+    @users('admin')
+    def test_activity_user_has_rights_for_upload(self):
+        group = self.env['res.groups'].create({'name': 'Restricted Access Group'})
+        user = self.env['res.users'].create({
+            'name': 'Restricted User',
+            'login': 'restricted_user',
+            'groups_id': [Command.set(group.ids)],
+        })
+        self.env['ir.model.access'].create({
```

Comment:
Don't think you need all that, there are simple rules based on 'access' field normally, could you check to simplify ?